api/desecapi/views/records.py

```diff
@@ -140,3 +147,18 @@ def get_serializer(self, *args, **kwargs):
     def perform_create(self, serializer):
         with PDNSChangeTracker():
             super().perform_create(serializer)
+
+
+class ZonefileView(DomainViewMixin, generics.RetrieveAPIView):
```

Comment:
I realize only now that this could be an `@action`-decorated method on the `DomainViewSet`, in which case things would fit more neatly into the DRF structure, and also the domain object would be readily available (via `self.get_object()`), so you wouldn't need to do the `DomainViewMixin` magic.

I have a slight preference for going this way, because (1) it saves code duplication (e.g. `get_queryset()`) and (2) is easier to maintain (e.g. we won't forget adding 2FA permissions here when merging 2FA functionality, as the `DomainViewSet`'s permissions would apply).

Reply:
... just realized that we already fell into that trap: `ZonefileView` is still missing at `throttle_scope` :)

Reply:
I've added a fixup to set the throttle scope correctly. For using the `@action` decorator, I think it is more efficient if you do that in the 2FA PR.

Reply:
Permissions were inconsistent with those for `GET rrsets/`. While adjusting that, I also moved the functionality to an `@action` decorator-style view.